libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts

```diff
@@ -173,8 +173,9 @@ export function useCustomPath(
       uri.searchParams.append("ajs_recover_source", source);
       uri.searchParams.append("ajs_recover_campaign", deeplinkCampaign);
     }
+
     return uri;
-  }, [deeplinkCampaign, modelUri, page, source]);
+  }, [deeplinkCampaign, page, servicesConfig?.params?.protectId, source]);
 
   return usePath(servicesConfig, customUri.toString()) ?? undefined;
```

Comment:
You can also remove this part as `usePath` already returns undefined if the feature is not enabled 
```suggestion
  return usePath(servicesConfig, customUri.toString());
```

Reply:
https://github.com/LedgerHQ/ledger-live/pull/5466/commits/07d8f158f2ad0c23fa058ee484fc1c63525720cc